source/SonicHeroes.WidescreenFix/dllmain.cpp

```diff
@@ -121,8 +116,45 @@ void Init()
                 fmul    dword ptr[eax08]
             }
         }
-    }; injector::MakeInline<CutOffAreaHook>(pattern.count(1).get(0).get<uint32_t>(0), pattern.count(1).get(0).get<uint32_t>(17));
-
+    }; injector::MakeInline<CutOffAreaHORHook>(pattern.count(1).get(0).get<uint32_t>(0), pattern.count(1).get(0).get<uint32_t>(17));
+    
+    struct CutOffAreaVertHook
+	{
+		void operator()(injector::reg_pack& regs)
+		{
+            float edx6C = *(float*)(regs.edx + 0x6C) / Screen.fHudZoomScale;
+            float eax10 = *(float*)(regs.eax + 0x10);
+            _asm
+            {
+                fld     dword ptr[edx6C]
+                fmul    dword ptr[eax10]
+            }
+         }
+    }; injector::MakeInline<CutOffAreaVertHook>(pattern.count(1).get(0).get<uint32_t>(42), pattern.count(1).get(0).get<uint32_t>(6));
+    
+    pattern = hook::pattern("83 C3 0C 53 56 E8 ?? ?? ?? ?? ??"); //0x6B7272
+    struct ShadowFixHook
+	{
+		void operator()(injector::reg_pack& regs)
+		{
+            float ebx00 = *(float*)(regs.ebx + 0x00);
+            float ebx04 = *(float*)(regs.ebx + 0x04);
+            _asm
+            {
+                add ebx,0x0C
+                fld     dword ptr[ebx00]
+                fmul    dword ptr[Screen.fHudScale]
+                fstp    dword ptr[ShadowFix]
+                fld     dword ptr[ebx04]
+                fmul    dword ptr[Screen.fHudZoomScale]
+                fstp    dword ptr[ShadowFix + 0x4]
+                mov ebx, ShadowFix
+                push ebx
```

Comment:
It is not a good idea to modify stack like this inside MakeInline. Better to make a hook without push/pop instructions.